cypress/e2e/gui/e2e.cy.js

```diff
@@ -0,0 +1,139 @@
+  });
+  
+  it('Mantém os filtros ao voltar da visualização de detalhes do cliente', () => {
+    cy.get('[data-testid="size-filter"]').select('Medium');
+    cy.get('[data-testid="industry-filter"]').select('Technology');
+    cy.contains('button', 'View').click();
+    cy.contains('button', 'Back').click();
+    cy.get('[data-testid="size-filter"]').should('have.value', 'Medium');
+    cy.get('[data-testid="industry-filter"]').should('have.value', 'Technology');
+  });
+
+  it('Retorna à lista de clientes ao clicar no botão "Voltar"', () => {
+    cy.get('[data-testid="size-filter"]').select('Medium');
+    cy.get('[data-testid="industry-filter"]').select('Technology');
+    cy.contains('button', 'View').click();
+    cy.contains('button', 'Back').click();
+    cy.contains('[data-testid="table"]', 'Below is our customer list.')
+      .should('be.visible');
+  });
+
+  it('Exibir o rodapé com o texto e links corretos', () => {
+    cy.contains('p', 'Copyright 2024 - Talking About Testing')
+      .should('be.visible');
+    cy.contains('a', 'Hotmart')
+      .should('be.visible')
+      .and('have.attr', 'href', 'https://hotmart.com/pt-br/club/cypress-playground-ate-a-nuvem');
+    cy.contains('a', 'Udemy')
+      .should('be.visible')
+      .and('have.attr', 'href', 'https://udemy.com/user/walmyr');
+    cy.contains('a', 'Blog')
+      .should('be.visible')
+      .and('have.attr', 'href', 'https://talkingabouttesting.com');
+    cy.contains('a', 'YouTube')
+      .should('be.visible')
+      .and('have.attr', 'href', 'https://youtube.com/@talkingabouttesting');
+  });
+
+  it('Exibe a saudação "Hi there" quando nenhum nome é fornecido', () => {
+    cy.contains('h2', 'Hi there!');
+  });
+
+  it('Exibe a saudação "Hi Joao" quando o nome é fornecido', () => {
+    cy.get('[data-testid="name"]').type('Joao');
+    cy.contains('h2', 'Hi Joao!');
+  });
+
+  it('Exibe o cabeçalho com um título, alternador de tema e um campo de entrada de texto', () => {
+    cy.get('[class^="Header_container"]').should('contain', 'EngageSphere');
+    cy.get('[class^="ThemeToggle_button"]').should('be.visible');
+    cy.get('[data-testid="name"]').should('be.visible');
+  });
+
+  it('Abre e fecha o messenger', () => {
+    cy.get('[class^="Messenger_openCloseButton"]')
+      .should('have.attr', 'aria-label', 'Open messenger');
+    cy.get('[class^="Messenger_openCloseButton"]').click();
+    cy.get('[class^="Messenger_openCloseButton"]')
+      .should('have.attr', 'aria-label', 'Close messenger');
+    cy.get('[class^="Messenger_openCloseButton"]').click();
+    cy.get('[class^="Messenger_openCloseButton"]')
+      .should('have.attr', 'aria-label', 'Open messenger');
+  });
+
+  it('Garante que todos os campos do messenger são obrigatórios e que o primeiro está focado', () => {
+    cy.get('[class^="Messenger_openCloseButton"]').click();
+    cy.get('button[type="submit"]').click();
+    cy.get('#messenger-name').should('have.attr', 'required');
```

Comment:
Aqui você já poderia estar verificando o foco também, sem a necessidade da verificação na linha 77.

```suggestion
    cy.get('#messenger-name')
      .should('be.focused')
      .and('have.attr', 'required')
```